packages/mui-base/src/InputUnstyled/InputUnstyled.test.tsx

```diff
@@ -31,6 +31,12 @@ describe('<InputUnstyled />', () => {
     expect(screen.getByRole('textbox')).to.have.tagName('textarea');
   });
 
+  it('should be able to attach input ref passed through props', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+    const { getByRole } = render(<InputUnstyled slotProps={{ input: { ref: inputRef } }} />);
+    expect(getByRole('textbox')).to.deep.equal(inputRef.current);
```

Comment:
Just to make sure input element is attached to ref

Reply:
We could use one more test to verify the behavior of useInput alone. See https://github.com/mui/material-ui/pull/35880/files#r1082379077

Reply:
Done